### PR TITLE
connected full-stack socket.io

### PR DESCRIPTION
**Pull request by ncen5293**
-Authenticated each connection
-only connect once if the user is logged in
-used cookies to save userId for each socket

### CODE REVIEW

**Review by aeciorc — CHANGES_REQUESTED** [review]
-Use a <script> index to load the socket client instead of installing it as a package.
-You should disconnect in the componentWillUnmount method and when user logs out

File: server/socket/socket.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const getDecodedToken = (socket) => {
+  const cookies = cookie.parse(socket.handshake.headers.cookie);
+  const token = cookies.token;
+  const decodedToken = jwt.verify(token, process.env.SECRET);
+  return decodedToken;
+}
+
+io.use((socket, next) => {
+  if (socket.handshake.headers && socket.handshake.headers.cookie) {
+    if (getDecodedToken(socket)) {
```
Comment (aeciorc) [inline server/socket/socket.js]: Looks like you're decoding it twice

File: server/socket/socket.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const getDecodedToken = (socket) => {
+  const cookies = cookie.parse(socket.handshake.headers.cookie);
+  const token = cookies.token;
+  const decodedToken = jwt.verify(token, process.env.SECRET);
+  return decodedToken;
+}
+
+io.use((socket, next) => {
+  if (getDecodedToken(socket)) {
```
Comment (aeciorc) [inline server/socket/socket.js]: oh, what I meant is that maybe you should associate the user id with the socket in this middleware

Comment (ncen5293) [reply server/socket/socket.js]: Alright, I changed it to put the userId in the middleware

**Comment by aeciorc** [comment]
Looks good! Let's get that merged

**Pull request merged**